README.md

```diff
@@ -84,18 +84,33 @@ Once you have implemented the tools, fill out the report below.
 
 ### Insights you may have had while implementing and comparing the algorithms. 
 
-*Describe this here.*
+Implementation wise, C is a little difficult to get started with.
+
+If you are used to just calling something like str.replace("\n",""),
+that is unfortunately not going to fly in C.
+
+Got the hang of it though, and playing around with pointers isn't so scary anymore.
 
 ### Problems encountered if any. 
 
-*Describe this here.*
+We malloc a lot in our parser, as we do not know how many pairs of fasta heads and sequences our file will contain.
+
+The solution seems to be to malloc n, then n *2, then n *4 etc., however it seems quite tedious to implement (and a little slower depending on input).
```

Comment:
Reallocation is done with `realloc()`. Don't use `malloc()` for that. It gets the job done, but less efficiently and you have to keep track of special cases yourself.

In any case, you can pack the strings into the buffer you load them into, so it isn't actually necessary to allocate new memory in the parsers, beyond what you need for loading the files. (It is a little bit for the reads, but you can reuse a buffer between each read you load, so it is not much of a bother).